tests/bgp/test_bgp_update_timer.py

```diff
@@ -305,3 +310,58 @@ def is_neighbor_sessions_established(duthost, neighbors):
         n1.stop_session()
         for route in constants.routes:
             duthost.shell("ip route flush %s" % route["prefix"])
+            
+
+def test_bgp_update_timer_session_down(
+    common_setup_teardown, constants, duthosts, enum_rand_one_per_hwsku_frontend_hostname,
+    toggle_all_simulator_ports_to_enum_rand_one_per_hwsku_frontend_host_m):   # noqa F811
+    
+    duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+
+    n0, n1 = common_setup_teardown
+    try:
+        n0.start_session()
+        n1.start_session()
+        
+        # ensure new sessions are ready
+        if not wait_until(WAIT_TIMEOUT, 5, 20, lambda: is_neighbor_sessions_established(duthost, (n0, n1))):
+            pytest.fail("Could not establish bgp sessions")
+
+        withdraw_intervals = []
+        for _, route in enumerate(constants.routes):
+            withdraw_intervals.append(-1)
+            n0.announce_route(route)
+            time.sleep(constants.sleep_interval)
+            dut_route = duthost.get_route(route["prefix"], n0.namespace)
+            if not dut_route:
+                pytest.fail("announce route %s from n0 to dut failed" % route["prefix"])
+        # close bgp session n0, monitor withdraw info from dut to n1
+        bgp_pcap = BGP_DOWN_LOG_TMPL
+        with log_bgp_updates(duthost, "any", bgp_pcap, n0.namespace):
+            duthost.shell("config bgp shutdown neighbor {}".format(n0.name))
+            global current_time 
+            current_time = time.time()        
+
+        with tempfile.NamedTemporaryFile() as tmp_pcap:
+                duthost.fetch(src=bgp_pcap, dest=tmp_pcap.name, flat=True)
+                duthost.file(path=bgp_pcap, state="absent")
+                bgp_updates = bgp_update_packets(tmp_pcap.name)
+                
+        for bgp_update in bgp_updates:
+            for i, route in enumerate(constants.routes):
+                if match_bgp_update(bgp_update, n1.peer_ip, n1.ip, "withdraw", route):
+                    withdraw_intervals[i] = bgp_update.time - current_time
+        
+        for i, route in enumerate(constants.routes):
+            if withdraw_intervals[i] >= constants.update_interval_threshold:
+                pytest.fail("withdraw updates interval %d exceeds threshold %d" % (constants.update_interval_threshold))
```

Comment:
There are two %d, but only one variable, pls use black or other tools to check the format.  @jcaiMR